subsys/bluetooth/host/hci_core.c

```diff
@@ -4020,7 +4020,7 @@ void bt_data_parse(struct net_buf_simple *ad,
 		}
 
 		if (len > ad->len) {
-			BT_WARN("Malformed data");
+			BT_DBG("malformed data");
```

Comment:
`bt_data_parse()` doesn't do any error reporting to the caller, so we'd be potentially suppressing important information to the application developer with the downgrade from WARN to DBG.

The general principle that we've tried to follow in the stack is that we use BT_WARN() whenever we receive something from a peer that we consider invalid or that we don't support, but which we can nevertheless handle gracefully (ignore or send an error PDU response) BT_ERR() on the other hand is used for local failures. What we can do, instead of changing the log level, is to better word the content of these log messages, to make it clear what they're about and that way avoid worrying the app developer. Btw, this comment applies to all other places as well in this PR where you're downgrading WARN to DBG for invalid or unsupported peer data. 

As for the placement of the implementation of `bt_data_parse()` I'm fine with moving it to a better place. Most likely it went into `hci_core.c` originally, since there wasn't any other more suitable "utilities" module back then (is there now?)